[Playback 2025] - Prompts

SergioEstevao opened the pull request:
| 📘 Part of: # |  <!-- project issue number, if applicable -->
|:---:|

Fixes PCIOS-178 <!-- issue number, if applicable -->
 
| 1 | 2 |
| - | - |
| <img width="750" height="1334" alt="Simulator Screenshot - iPhone SE (2nd generation) - 2025-10-07 at 14 15 30" src="https://github.com/user-attachments/assets/b4aea0f0-a04d-4a0b-ada7-bf7eca0bbeaf" /> | <img width="750" height="1334" alt="Simulator Screenshot - iPhone SE (2nd generation) - 2025-10-07 at 14 27 30" src="https://github.com/user-attachments/assets/d92506eb-416d-478f-b2cf-edad5c51f570" /> |

<!-- Please include a summary of what this PR is changing and why these changes are needed. -->
This PR implements the FF for the 2025 end of year playback, and the two entry points in the app.

At the moment this PR links to the templates stories of 2024, but we can do in follow up PR an update to each story, because they are sharing the same data as last year.

## To test

- Start the app
- Login with an user that has some stats for 2025
- Go to Profile -> Settings -> Beta features
- Enable the FF "End of Year 2025"
- Go to Developer Menu and tap on "End Of Year -> Reset modal/profile badge" and select 2025
- Restart the app
- Check that if you see the prompt for the End of the Year on the Podcasts screen
- Tap on View My Playback 2025
- It should show the 2024 design but with the stats for 2025 ( some screens for 2024 had the year hardcoded on string/images so ignore those)
- Go to Profile 
- Check that you see a banner for the playback 2025
- Tap on it
- Check if Playback screen shows correctly

## Checklist

- [ ] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

SergioEstevao committed:
Add FF for end of year 2025

SergioEstevao committed:
Add playback 2025 flow

SergioEstevao committed:
Update strings for end of year 2025

SergioEstevao committed:
Update prompt screen messages

danielebogo reviewed (approved):
All good @SergioEstevao !
Prompt appeared when I launched the app and I can see the banner in Profile

SergioEstevao committed:
Fix query for top podcasts

When using GRDB the episode uuid field was given priority over the podcast.uuid field so we were getting the wrong uuids inside the Podcast result
